api/v1alpha1/clusterextension_types.go

```diff
@@ -46,6 +46,78 @@ const (
+	//
+	// installNamespace is required, immutable, and follows the DNS label standard
+	// as defined in RFC 1123. This means that valid values:
+	//   - Contain no more than 63 characters
+	//   - Contain only lowercase alphanumeric characters or '-'
+	//   - Start with an alphanumeric character
+	//   - End with an alphanumeric character
+	//
+	// Some examples of valid values are:
+	//   - some-namespace
+	//   - 123-namespace
+	//   - 1-namespace-2
+	//   - somenamespace
+	//
+	// Some examples of invalid values are:
+	//   - -some-namespace
+	//   - some-namespace-
+	//   - thisisareallylongnamespacenamethatisgreaterthanthemaximumlength
+	//   - some.namespace
+	//
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	//+kubebuilder:validation:MaxLength:=63
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="installNamespace is immutable"
+	InstallNamespace string `json:"installNamespace"`
+
+	// preflight is an optional field that can be used to configure the preflight checks run before installation or upgrade of the content for the package specified in the packageName field.
+	//
+	// When specified, it overrides the default configuration of the preflight checks that are required to execute successfully during an install/upgrade operation.
+	//
+	// When not specified, the default configuration for each preflight check will be used.
+	//
+	//+optional
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+
+	// serviceAccount is a required reference to a ServiceAccount that exists
+	// in the installNamespace. The provided ServiceAccount is used to install and
+	// manage the content for the package specified in the packageName field.
+	//
+	// In order to successfully install and manage the content for the package,
+	// the ServiceAccount provided via this field should be configured with the
+	// appropriate permissions to perform the necessary operations on all the
+	// resources that are included in the bundle of content being applied.
+	ServiceAccount ServiceAccountReference `json:"serviceAccount"`
+}
+
+const ResolutionTypeCatalog = "Catalog"
+
+// SourceConfig is a discriminated union which selects the type of resolver.
+// +union
+// +kubebuilder:validation:XValidation:rule="self.sourceType == 'Catalog' && has(self.catalog)",message="sourceType Catalog requires catalog field"
+type SourceConfig struct {
+
+	// sourceType selects the type of resolver and currently supports only one value: "Catalog"
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Catalog"
+	// +kubebuilder:validation:Required
+	SourceType string `json:"sourceType,omitempty"`
+
+	// catalog defines required fields for catalog resolution. This field must be defined when sourceType is set to "Catalog".
+	// +optional.
+	Catalog *CatalogResolution `json:"catalog,omitempty"`
+}
+
+// CatalogResolution defines the required fields for catalog source.
+type CatalogResolution struct {
```

Comment:
Consistency in naming:
```suggestion
// CatalogSource defines the required fields for catalog source.
type CatalogSource struct {
```

Reply:
This change is reflected in the latest squash commit.